File: go/tutorials/tour-of-restate-go/app/ticketobject.go
package main

import restate "github.com/restatedev/sdk-go"

type TicketObject struct{}

func (TicketObject) Reserve(ctx restate.ObjectContext) (bool, error) {
	return true, nil
}

func (TicketObject) Unreserve(ctx restate.ObjectContext) error {
	return nil
}

func (TicketObject) MarkAsSold(ctx restate.ObjectContext) error {
	return nil
}

File: typescript/tutorials/tour-of-restate-typescript/src/app/ticket_object.ts
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate examples,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

import * as restate from "@restatedev/restate-sdk";
import { TicketStatus } from "../auxiliary/ticket_status";

export const ticketObject = restate.object({
  name: "TicketObject",
  handlers: {
    async reserve(ctx: restate.ObjectContext) {
      return true;
    },

    async unreserve(ctx: restate.ObjectContext) {
      return;
    },

    async markAsSold(ctx: restate.ObjectContext) {
      return;
    },
  },
});

export const TicketObject: typeof ticketObject = { name: "TicketObject" };
